ngStorage.js

```diff
@@ -97,15 +108,42 @@
                         },
                         $reset: function(items) {
                             for (var k in $storage) {
-                                '$' === k[0] || (delete $storage[k] && webStorage.removeItem('ngStorage-' + k));
+                                '$' === k[0] || (delete $storage[k] && webStorage.removeItem(STORAGE_PREFIX + k));
                             }
 
                             return $storage.$default(items);
                         },
                         $sync: function () {
-                            for (var i = 0, l = webStorage.length, k; i < l; i++) {
+                            var i = webStorage.length,
+                                $storageKey,
+                                k;
+
+                            while (i--) {
+                                k = webStorage.key(i);
+                                $storageKey = getStorageKey(k);
```

Comment:
note that there's a problem here, the test for k needs to be moved up before the getStorageKey call as otherwise .split() will throw an error if k is undefined